File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<String> res=new ArrayList<>();
        int j=0;
        for(int i=1;i<=n&&j<target.length;i++){
            res.add("Push");
            if(i==target[j])
                j++;
            else
                res.add("Pop");
        }
        return res;
    }
}

File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.py
class Solution:
    def buildArray(self, target: list[int], n: int) -> list[str]:
        j=0
        i=1
        res=[]
        while i<=n and j<len(target):
            res.append("Push")
            if i==target[j]:
                j+=1
            else:
                res.append("Pop")
            i+=1
        return res